### DISCUSSION TITLE
How do I use parent route loader data in child route component?

### DISCUSSION THREAD

**Discussion by Silverwolf90** [TanStack/router#979] [Q&A]
Every child has the same loader, so I figured I could hoist this loader into the parent's route and access in each child. But I can't seem to figure out how to access this data in simple way/type safe way. Is there a straightforward way to do this out-of-the-box? Or would expected way to wrap an `Outlet` in a react context that contains the loader data and access the context from children?

I have tried using the following lines in the child route's component, but I get an "Invariant failed" runtime error in both cases:

```
const data = useLoaderData({ from: parentRoute.id });
```

```
const data = parentRoute.useLoaderData();
```

**Comment by SeanCassiere** [comment]
Would need a reproduction, but off the bat this should work.

That being said, I don't try to use `parentRoute.id` accessors, instead I stick to using "strings" that anyways get validated by `tsc`.

**Reply by SeanCassiere** [reply]
Just confirmed that it does indeed work.

![image](https://github.com/user-attachments/assets/01287621-3215-46c8-8d63-5096b611e750)

![image](https://github.com/user-attachments/assets/c5d34007-95ed-4f78-af07-98a846fe7206)

**Reply by SeanCassiere** [reply]
You'd only be getting an error thrown of if the parent route isn't currently in the render stack.
This logically wouldn't work, since you can't access `loaderData` which hasn't been fetched.
